components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7713,6 +7709,29 @@ private boolean deleteComment(String uuid, String commentId, Connection connecti
         return false;
     }
 
+    private void deleteAPIComments(String uuid, Connection connection) throws APIManagementException {
+        int id = -1;
+        ResultSet resultSet = null;
+        try {
+            id = getAPIID(uuid, connection);
+            if (id == -1) {
+                String msg = "Could not load API record for API with UUID: " + uuid;
+                throw new APIManagementException(msg, ExceptionCodes.from(ExceptionCodes.API_NOT_FOUND, uuid));
+            }
+            connection.setAutoCommit(false);
+            String getParentCommentIdsQuery = SQLConstants.GET_PARENT_COMMENT_IDS_FOR_API;
```

Comment:
do we need to retrieve parent comments

Reply:
since we can run same query on DELETE
 "DELETE AM_API_COMMENTS.COMMENT_ID "
                    + "FROM "
                    + "AM_API_COMMENTS "
                    + "WHERE "
                    + "API_ID = ? "
                    + "AND PARENT_COMMENT_ID = NULL";

Reply:
Changes made